src/app/shared/common-utils/common-utils.ts

```diff
@@ -14,4 +14,16 @@ export const formatNumber = (number, suffix = '', format = NUMERAL_FORMAT) => {
         return nNumeral.format(format) + suffix;
     }
     return number;
-};
\ No newline at end of file
+};
+
+export function isWeekday(dateValue): boolean {
+    const formatedDate = moment(dateValue).format(DATE_FORMAT);
+    const date = moment(formatedDate, DATE_FORMAT, true);
```

Comment:
Why not to do just: `` const date = moment(dateValue) `` ? Currently, you are using relaxed parsing, create a formated date; read formated date in a strict parsing (what should) and get moment internal date representation.

Reply:
Yes, as I don't know the exact date formate coming from API. I just convert it according to the logic. If you want me to make it strict to the date format then I will do it. Please confirm.

Reply:
Okay Fixed